src/09_output_housing_value.R

```diff
@@ -0,0 +1,466 @@
+  folio_data <- bca_properties %>%
+    left_join(bca_addrs, by = "FOLIO_ID") %>%
+    left_join(gcs_data, by = c("PC" = "POSTALCODE"))
+
+  # Step 3: Calculate average property statistics by DA and jurisdiction
+  ave_p <- folio_data %>%
+    group_by(YR, DA) %>%
+    summarize(
+      N_PROPERTY = n_distinct(FOLIO_ID),
+      AVE_IMPROVEMENT_VALUE = mean(GEN_GROSS_IMPROVEMENT_VALUE, na.rm = TRUE),
+      AVE_LAND_VALUE = mean(GEN_GROSS_LAND_VALUE, na.rm = TRUE),
+      AVE_TOTAL_VALUE = mean(TOTAL_VALUE, na.rm = TRUE),
+      MIN_TOTAL_VALUE = min(TOTAL_VALUE, na.rm = TRUE),
+      MAX_TOTAL_VALUE = max(TOTAL_VALUE, na.rm = TRUE),
+      .groups = "drop"
+    )
+
+  ave_p <- ave_p %>%
+    collect()
+
+  # ave_p %>% count(DA) %>% filter(n > 1)
+  # ave_p %>%
+  #   filter(DA == "010120219")
+  # 1063 DAs in 2023 has at least two JURISDICTIONs. It causes another problem.
+
+  # Step 4: Calculate median property values
+  med_p <- folio_data %>%
+    group_by(YR, DA) %>%
+    mutate(
+      MEDIAN_TOTAL_VALUE = median(TOTAL_VALUE, na.rm = TRUE)
+    ) %>%
+    distinct(YR, DA, MEDIAN_TOTAL_VALUE) %>%
+    collect()
+
+  # Store the year's results in the list
+  # After calculating ave_p and med_p, directly append to combined dataframes
+  combined_ave_p <- bind_rows(combined_ave_p, ave_p)
+  combined_med_p <- bind_rows(combined_med_p, med_p)
+}
+
+# Step 5: Get average median income by DA for 2021 tax year. There are many years of income data from 2016 to 2021. 2021 is the most recent year.
+# Links income data with geographic identifiers
+income_data <- tbl(
+  con,
+  sql(sprintf(
+    "
+    SELECT [tax_year],
+           [postal_code],
+           [median_income]
+    FROM [%s].[Prod].[%s] 
+    WHERE [tax_year]='2021'
+",
+    config$database$database,
+    income_table
+  ))
+)
+
+inc <- income_data %>%
+  left_join(gcs_data, by = c("postal_code" = "POSTALCODE")) %>%
+  group_by(tax_year, DA) %>%
+  summarize(
+    AVE_MED_INCOME = mean(median_income, na.rm = TRUE),
+    .groups = "drop"
+  )
+
+
+# Collect the income data
+inc_collected <- collect(inc)
+
+# Final step: Combine all statistics
+
+# Check if we have any data before continuing
+if (
+  !is.null(combined_ave_p) &&
+    nrow(combined_ave_p) > 0 &&
+    !is.null(combined_med_p) &&
+    nrow(combined_med_p) > 0
+) {
+  # Final step: Combine all statistics
+  final_data <- combined_ave_p %>%
+    left_join(combined_med_p, by = c("DA", "YR")) %>%
+    arrange(YR, DA)
+
+  final_data_with_income <- final_data %>%
+    left_join(inc_collected, by = "DA") %>%
+    arrange(YR, DA)
+
+  # Count records with and without income data
+  income_count <- sum(!is.na(final_data_with_income$AVE_MED_INCOME))
+  cat(glue("Total records: {nrow(final_data)}\n"))
+  cat(glue("Records with income data: {income_count}\n"))
+
+  # View the first few rows of the result
+  print(head(final_data_with_income))
+
+  # Save the final data to a CSV file with date and time in the filename
```

Comment:
I would move all the save steps to outside of the if statement so that if you don't want to write everything but you want to create the final data tables, you can without accidentally saving as you step through your code.